orttraining/orttraining/python/training/_ortmodule_io.py

```diff
@@ -64,33 +150,34 @@ def _replace_stub_with_tensor_value(user_output, outputs, output_idx):
 
     return user_output
 
-def _extract_output_schema(output):
-    """Extract the output schema by replacing every torch.Tensor value with _TensorStub"""
 
-    if output is None:
+def _extract_schema(data):
+    """Extract the data schema by replacing every torch.Tensor value with _TensorStub"""
+
+    if data is None:
         return None
-    # Depth first traversal to iterate over the output to replace every tensor with a stub
-    elif isinstance(output, torch.Tensor):
-        return _TensorStub()
-
-    if isinstance(output, abc.Sequence):
-        sequence_type = type(output)
-        output = list(output)
-        for idx in range(len(output)):
-            output[idx] = _extract_output_schema(output[idx])
+    # Depth first traversal to iterate over the data to replace every tensor with a stub
+    elif isinstance(data, torch.Tensor):
+        return _TensorStub(None, str(data.dtype), None, len(data.size()))
```

Comment:
We should probably explicitly set the attributes using assignment (instead of passing arguments positionally).